### ISSUE TITLE
Different versions of the same artifact get conflated

### ISSUE THREAD

**Issue by yrashk** [lburgazzoli/gradle-karaf-plugin#34]
I have a dependency on `com.google.guava:guava:20.0` and some other dependencies have a transitive dependency on `com.google.guava:guava:19.0`. When I generate a feature file, I only get this:

```xml
<bundle>mvn:com.google.guava/guava/20.0</bundle>
```

Is there any way we can support multiple versions of the artifact to satisfy the dependencies? After all, isn't OSGi supposed to be a solution to this problem in the first place :)

**Comment by lburgazzoli** [comment]
Can you provide an example fir a lazy man?

**Comment by yrashk** [comment]
This combination: 

```groovy
compile 'com.graphql-java:graphql-java-servlet:0.9.0'
compile com.google.guava:guava:20.0
```

will carry both 19.0 (from graphql-java-servlet) and 20.0 (explicit)

Does this help?

**Comment by yrashk** [comment]
So what seemingly worked for now is adding a special configuration (`extra`) and doing

```groovy
extra 'com.google.guava:guava:19.0'
// ...
configurations 'compile', 'extra'
```

but this is extremely unobvious

**Comment by lburgazzoli** [comment]
I think the issue is that guava:19.0 is "promoted" to guava:20.0:

```
+--- com.graphql-java:graphql-java-servlet:0.9.0
|    +--- org.slf4j:slf4j-api:1.7.21
|    +--- com.google.guava:guava:19.0 -> 20.0
|    +--- org.projectlombok:lombok:1.16.8
|    +--- javax.servlet:javax.servlet-api:3.0.1
|    +--- commons-fileupload:commons-fileupload:1.3.1
|    |    \--- commons-io:commons-io:2.2
|    +--- com.graphql-java:graphql-java:2.2.0
|    |    +--- org.antlr:antlr4-runtime:4.5.1
|    |    +--- org.slf4j:slf4j-api:1.7.12 -> 1.7.21
|    |    \--- org.antlr:antlr4:4.5.1
|    +--- com.graphql-java:graphql-java-annotations:0.13.0
|    |    +--- javax.validation:validation-api:1.1.0.Final
|    |    +--- com.graphql-java:graphql-java:2.2.0 (*)
|    |    \--- org.projectlombok:lombok:1.16.6 -> 1.16.8
|    +--- com.fasterxml.jackson.core:jackson-core:2.7.3
|    +--- com.fasterxml.jackson.core:jackson-annotations:2.7.3
|    +--- com.fasterxml.jackson.core:jackson-databind:2.7.3
|    |    +--- com.fasterxml.jackson.core:jackson-annotations:2.7.0 -> 2.7.3
|    |    \--- com.fasterxml.jackson.core:jackson-core:2.7.3
|    \--- com.fasterxml.jackson.datatype:jackson-datatype-jdk8:2.7.3
|         +--- com.fasterxml.jackson.core:jackson-core:2.7.3
|         \--- com.fasterxml.jackson.core:jackson-databind:2.7.3 (*)
\--- com.google.guava:guava:20.0
```

Need to check how the code handles it right now.

**Comment by lburgazzoli** [comment]
So yes, the problem is that gradle resolves the conflict and promotes guava:20.0 so in the resolved dependencies I do not have a clear reference to guava:19.0, I would need to check the resolution for each dependency and check if it has a conflict one and eventually take it into account.

Then the problem is that gradle the discarded dep does not have an associated file so it may make it more complicated to create a kar, need to think about it.

Any suggestion would be very welcome.

**Comment by lburgazzoli** [comment]
For the moment the only way to make it working is to use a special configuration for the extra stuffs
